nucypher/blockchain/eth/interfaces.py

```diff
@@ -474,11 +475,21 @@ def get_contract_by_name(self,
                 except IndexError:
                     raise self.UnknownContract(f"There are no Dispatcher records targeting '{name}'")
 
-        else:  # It's not upgradeable
+        else:
+            # NOTE: 0 must be allowed as a valid version number
             if len(target_contract_records) != 1:
-                m = "Multiple records registered for non-upgradeable contract {}"
-                raise self.InterfaceError(m.format(name))
-            _target_contract_name, selected_address, selected_abi = target_contract_records[0]
+                if version is None:
+                    m = f"{len(target_contract_records)} records enrolled for contract {name} " \
+                        f"and no version index was supplied."
+                    raise self.InterfaceError(m.format(name))
```

Comment:
`name` already provided in f-string.

```suggestion
                    raise self.InterfaceError(m)
```

Reply:
@KPrasch ^ missed one - probably got lost in the shuffle 🙂

Reply:
:heavy_check_mark: